spec/system/admin/reports/sales_tax/sales_tax_totals_by_order_spec.rb

```diff
@@ -153,6 +153,82 @@
         "order1@example.com"
       ].join(" "))
     end
+
+    context "with a voucher" do
+      let(:voucher) do
+        create(:voucher_flat_rate, code: 'some_code', enterprise: distributor, amount: 10)
+      end
+
+      before do
+        Flipper.enable :vouchers
+
+        # Add voucher to the order
+        voucher.create_adjustment(voucher.code, order)
+        VoucherAdjustmentsService.new(order).update
+
+        # the enterprise fees can be known only when the user selects the variants
+        # we'll need to create them by calling recreate_all_fees!
+        order.recreate_all_fees!
+
+        order.update_totals_and_states
+        OrderWorkflow.new(order).complete!
+      end
+
+      pending "generates the report with the correct tax amount" do
+        visit_sales_tax_totals_by_order
+
+        expect(page).to have_button("Go")
+        click_on "Go"
+        expect(page.find("table.report__table thead tr").text).to have_content(table_header)
+
+        # Calculation details :
+        # Price excl tax = order.total - total_tax + voucher tax (negative number)
+        # Tax amount = Non adjusted tax - voucher tax
+        # Price incl tac = Price excl tax + non adjusted tax
+        expect(page.find("table.report__table tbody").text).to have_content([
+          "DistributorEnterpriseWithTax",
```

Comment:
Does this need to be a system spec? Could we test this on the lib level? It would be a lot faster and maybe more convenient to have the proper arrays already.

Reply:
We could probably test this at the lib level. I am not really familiar with report, so it was easier to plug my scenario in the existing system spec. Do you think I should move the test out of the system spec ?

Reply:
> Do you think I should move the test out of the system spec ?

Yes, I think that would be good.